include/quicr/track_name.h

```diff
@@ -3,12 +3,110 @@
+    {
+      public:
+        TrackNamespace() = default;
+
+        template<typename... B,
+                 std::enable_if_t<std::is_same_v<std::common_type_t<B...>, std::vector<uint8_t>>, bool> = true>
+        explicit TrackNamespace(B&&... entries)
+        {
+            static_assert(sizeof...(B) >= 1, "Track namespace must have at least 1 entry");
+            static_assert(sizeof...(B) <= 32 - 1, "Track namespace can only have a maximum of 32 entries");
+
+            std::size_t offset = 0;
+            const auto add_entry = [&](auto&& e) {
+                entries_.emplace_back(Span{ bytes_ }.subspan(offset, e.size()));
+                offset += e.size();
+            };
+
+            (bytes_.insert(bytes_.end(), entries.begin(), entries.end()), ...);
+            (add_entry(entries), ...);
+        }
+
+        TrackNamespace(const std::vector<std::vector<uint8_t>>& entries)
+        {
+            if (entries.size() > 32 || entries.size() == 0) {
+                throw std::invalid_argument("TrackNamespace requires a number of entries in the range of [1, 32]");
+            }
+
+            for (auto& entry : entries) {
+                bytes_.insert(bytes_.end(), entry.begin(), entry.end());
+            }
+
+            std::size_t offset = 0;
+            for (auto& entry : entries) {
+                entries_.emplace_back(Span{ bytes_ }.subspan(offset, entry.size()));
+                offset += entry.size();
+            }
+        }
+
+        TrackNamespace(const TrackNamespace&) = default;
+        TrackNamespace(TrackNamespace&&) = default;
+
+        TrackNamespace& operator=(const TrackNamespace&) = default;
+        TrackNamespace& operator=(TrackNamespace&&) = default;
+
+        const std::vector<Span<const uint8_t>>& GetEntries() const noexcept { return entries_; }
+
+        // NOLINTBEGIN(readability-identifier-naming)
+        auto begin() noexcept { return bytes_.begin(); }
+        auto end() noexcept { return bytes_.end(); }
+        auto begin() const noexcept { return bytes_.begin(); }
+        auto end() const noexcept { return bytes_.end(); }
+        auto data() const noexcept { return bytes_.data(); }
+        auto size() const noexcept { return bytes_.size(); }
+        bool empty() const noexcept { return bytes_.empty(); }
+        // NOLINTEND(readability-identifier-naming)
+
+        friend bool operator==(const TrackNamespace& lhs, const TrackNamespace& rhs) noexcept
+        {
+            return lhs.bytes_ == rhs.bytes_;
+        }
+
+        friend bool operator!=(const TrackNamespace& lhs, const TrackNamespace& rhs) noexcept { return !(lhs == rhs); }
+
+        friend bool operator<(const TrackNamespace& lhs, const TrackNamespace& rhs) noexcept
+        {
+            return lhs.bytes_ < rhs.bytes_;
+        }
+
+        friend bool operator>(const TrackNamespace& lhs, const TrackNamespace& rhs) noexcept
+        {
+            return lhs.bytes_ > rhs.bytes_;
+        }
```

Comment:
this is useful. can you please add few tests for comparing different things ( like you had for qnames) .. basically some form of prefix tests

Reply:
I'm not fully sure as of yet whether or not it does the kind of prefix checking we want, but it may work well for it. I'll add the tests.

Reply:
I have few thoughts that I would like to go through you tomorrow on matching at tuple boundaries. This might need adding boundary information in tuple elements.